src/Http/Routing/test/UnitTests/Constraints/FloatRouteConstraintTests.cs

```diff
@@ -13,7 +13,7 @@ public class FloatRouteConstraintTests
         [InlineData(3.14, true)]
         [InlineData("not-parseable-as-float", false)]
         [InlineData(false, false)]
-        [InlineData("1.79769313486232E+300", false)]
+        [InlineData("1.79769313486232E+300", true)] // Parses as infinity
```

Comment:
I think despite the change, this is the correct behaviour. You wanted a float. Infinity and Nan are floats. #YOLO

I think the feature as a whole has questionable practical value, I dunno why you'd want to serialize a float in the URL path.